### PR TITLE
[FLINK-18550][sql-client] use TableResult#collect to get select result for sql client

### PR DESCRIPTION
**Pull request by fsk119**
## What is the purpose of the change

*This pr aims to use TableResult#collect to get select result instead of using user defined sink. which could reduce a lot of code.*

## Brief change log

- *use TableResult#collect to get select result for sql client*
- *remove useless code about Deployment in sql client*
- *use RowKind instead of Tuple2 to represent changes in sql client*
- *fix data loss in ChangelogCollectResult*

## Verifying this change

*This change is already covered by existing tests, such as LocalExecutorITCase, and add some new test to verify the change, such as MaterializedCollectBatchResultTest, ChangelogCollectResultTest.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliChangelogResultView.java [hunk 1]
```diff
@@ -124,18 +125,11 @@ protected void refresh() {
...
                     // convert row
-                    final String[] changeRow = new String[change.f1.getArity() + 1];
-                    final String[] row = PrintUtils.rowToString(change.f1);
-                    System.arraycopy(row, 0, changeRow, 1, row.length);
-                    if (change.f0) {
-                        changeRow[0] = "+";
-                    } else {
-                        changeRow[0] = "-";
-                    }
+                    final String[] row = PrintUtils.rowToString(change, NULL_COLUMN, true);
```
Comment (wuchong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliChangelogResultView.java:132]: It seems that `PrintUtils.rowToString` doesn't correctly stringify values of TIMESTAMP type. I created FLINK-21456 to track this effort.

**Review by wuchong — COMMENTED** [review]
Thanks for the contribution @fsk119 . I left some comments.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliChangelogResultView.java (hunk 1 above)
Comment (fsk119) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliChangelogResultView.java:132]: Sure. I will take it.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java [hunk 2]
```diff
@@ -654,11 +654,7 @@ private void callSelect(SqlCommandCall cmdCall) {
         if (resultDesc.isTableauMode()) {
             try (CliTableauResultView tableauResultView =
                     new CliTableauResultView(terminal, executor, sessionId, resultDesc)) {
-                if (resultDesc.isMaterialized()) {
-                    tableauResultView.displayBatchResults();
-                } else {
-                    tableauResultView.displayStreamResults();
-                }
+                tableauResultView.displayResults(resultDesc.isStreamingMode());
```
Comment (wuchong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java:657]: The `CliTableauResultView` already holds the `resultDesc`, so there is no need to pass the streaming mode again.

Comment (wuchong) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java:657]: The `CliTableauResultView` already holds the `resultDesc`, so there is no need to pass the streaming mode again.

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliTableauResultViewTest.java [hunk 3]
```diff
@@ -286,7 +320,7 @@ public void testFailedBatchResult() {
 
     @Test
     public void testStreamingResult() {
-        ResultDescriptor resultDescriptor = new ResultDescriptor("", schema, true, true);
+        ResultDescriptor resultDescriptor = new ResultDescriptor("", schema, true, true, false);
```
Comment (wuchong) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/cli/CliTableauResultViewTest.java:323]: The last flag should be true.

**Review by wuchong — APPROVED** [review]
LGTM.

**Comment by wuchong** [comment]
Build is passed: https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=13740&view=results
Merging...

**Pull request merged**